collector/mdadm_linux.go

```diff
@@ -152,6 +152,8 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 	mdStates := make([]mdStatus, 0, estimateMDs)
 
 	for i, l := range lines {
+		active, total, size = 0, 0, 0
```

Comment:
If we can't reliably say something about some values, like `total`, I'd vote to either use floats+NaN or ints+`-1` and exchange the -1 to NaN in the caller.